PurrSoft/PurrSoft.Api/Controllers/AnimalController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PurrSoft.Api.Controllers.Base;
+using PurrSoft.Application.Commands.AnimalCommands;
+using PurrSoft.Application.Common;
+using PurrSoft.Domain.Entities;
+using System.Net;
+
+namespace PurrSoft.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class AnimalController : BaseController
+{
+    public AnimalController()
+    {
+    }
+
+    [HttpGet]
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    [ProducesResponseType(typeof(ICollection<Animal>), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(CommandResponse), (int)HttpStatusCode.BadRequest)]
+    public async Task<CollectionResponse<Animal>> GetAnimalsAsync()
```

Comment:
Also I believe you can put it as a parameter the command i mean . From body . So that you don't create it there.
Because if there is nothing in body it should be fine since AnimalGetCommand for now has no attributes.